### PR TITLE
Update why kedro module

### LINKED ISSUE #134: Update Why Kedro module
For the website rebrand, we need to update the `Why Kedro?` module to look like this: 

image

Here's the Figma file: https://www.figma.com/file/fJLMAWpBfh75loDtclpdXR/Kedro-Website?type=design&node-id=1601%3A16983&t=jD0WMhiXRO8seHFJ-1

**Acceptance criteria:**

- [x] When a user scrolls to this section, it should become sticky
- [x] When a user scrolls to this section and the section becomes sticky, the explanation text should scroll ([prototype](https://www.figma.com/proto/fJLMAWpBfh75loDtclpdXR/Kedro-Website?page-id=1650%3A6141&type=design&node-id=1741-6001&viewport=-5839%2C1800%2C0.3&scaling=min-zoom&starting-point-node-id=1741%3A6001&show-proto-sidebar=1))
- [x] When a user scrolls to this section and the section becomes sticky and the user reaches the end of the fourth reason why someone should use Kedro, the section is no longer sticky and the user continues to scroll down the page.

### PR DESCRIPTION
**Pull request by Huongg**
## Description

Resolves #134 

## Development notes

https://github.com/kedro-org/kedro-website/assets/32060364/9d3db50d-0f2b-419f-8630-d5c6ba73f755

NOTES:
- The gradient block currently only applies to the top of the text box. The reason for it is that in order to make the mouse scrollable anywhere on the "Why Kedro" section, I didn't set the fixed height/max height as in the design. Therefore, it's very hard to set the gradient block for the bottom of the section as there is no fixed height anymore. 
Let me know what you think on this @Mackay031 @stephkaiser? 

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Added tests to cover my changes, where applicable

### CODE REVIEW

File: modules/index/why-kedro/why-kedro.module.scss [hunk 1]
```diff
@@ -16,176 +16,128 @@
...
-  padding: 0 1rem 0.35rem;
-  position: relative;
-  transition: color $transition-time ease;
-  white-space: nowrap;
-  width: 100%;
-
-  &:first-of-type {
-    padding-left: 0;
+    text-align: center;
+    margin-top: 1rem;
```
Comment (ravi-kumar-pilla) [inline modules/index/why-kedro/why-kedro.module.scss:69]: [Nit] Can we have alphabetical order. Thank you

File: modules/index/why-kedro/why-kedro.module.scss [hunk 2]
```diff
@@ -16,176 +16,128 @@
...
-  margin: 0 auto 5rem;
-  max-width: 1000px;
-  min-height: 5.75rem;
+.contentBlock {
+  margin-bottom: 3rem;
+  width: 832px;
 
   @include breakpoint(medium-only) {
-    min-height: 7rem;
+    width: 100%;
```
Comment (ravi-kumar-pilla) [inline modules/index/why-kedro/why-kedro.module.scss:113]: [Nit] Can we have alphabetical order. Thank you

File: modules/index/why-kedro/why-kedro.module.scss [hunk 3]
```diff
@@ -16,176 +16,128 @@
...
     padding: 0 1rem;
   }
 
   @include breakpoint(small) {
-    margin-bottom: 3rem;
-    min-height: 9rem;
+    width: 100%;
```
Comment (ravi-kumar-pilla) [inline modules/index/why-kedro/why-kedro.module.scss:118]: [Nit] Can we have alphabetical order. Thank you

**Review by ravi-kumar-pilla — APPROVED** [review]
Everything else looks good to me. Thank you

File: modules/index/why-kedro/why-kedro.module.scss [hunk 4]
```diff
@@ -16,176 +16,128 @@
...
 }
 
-.list,
-.listItem {
-  list-style: none;
-  margin: 0;
-  padding: 0;
-}
+.title {
+  @include display-xl;
```
Comment (tynandebold) [inline modules/index/why-kedro/why-kedro.module.scss:60]: The `font-weight` for the title is off. Think it should be `500`.

File: modules/index/why-kedro/why-kedro.module.scss
Comment (tynandebold) [inline modules/index/why-kedro/why-kedro.module.scss:1]: Can you please update the `.outer` style padding to match the designs?

For the `medium-up` breakpoint I think it needs to be `padding: 7.5rem 3.5rem`.

**Review by tynandebold — APPROVED** [review]
Looks good! Some style comments from me. Please make sure those are fixed. Otherwise well done.

File: modules/index/why-kedro/why-kedro.module.scss [hunk 5]
```diff
@@ -16,176 +16,128 @@
...
   }
 }
 
-.tabContent {
-  @include h3;
-
-  margin: 0 auto 5rem;
-  max-width: 1000px;
-  min-height: 5.75rem;
+.contentBlock {
```
[the remark this answers is not in the document]
Comment (Huongg) [reply modules/index/why-kedro/why-kedro.module.scss:109]: yup done, ty

File: modules/index/why-kedro/why-kedro.module.scss
Comment (Huongg) [reply modules/index/why-kedro/why-kedro.module.scss:1]: good spot, thanks

**Pull request merged**